File: Backend in .NET/Models/Admission.cs
namespace SmartSchoolAPI.Models;

public class Admission
{
    public int AdmissionId { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public DateTime DateOfBirth { get; set; }
    public string Gender { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string State { get; set; } = string.Empty;
    public string Pincode { get; set; } = string.Empty;
    public string AppliedClass { get; set; } = string.Empty;
    public string AcademicYear { get; set; } = string.Empty;
    
    // Parent/Guardian Information
    public string FatherName { get; set; } = string.Empty;
    public string FatherOccupation { get; set; } = string.Empty;
    public string FatherPhone { get; set; } = string.Empty;
    public string MotherName { get; set; } = string.Empty;
    public string MotherOccupation { get; set; } = string.Empty;
    public string MotherPhone { get; set; } = string.Empty;
    public string GuardianName { get; set; } = string.Empty;
    public string GuardianRelation { get; set; } = string.Empty;
    public string GuardianPhone { get; set; } = string.Empty;
    
    // Previous School Information
    public string? PreviousSchool { get; set; }
    public string? PreviousClass { get; set; }
    public decimal? PreviousMarks { get; set; }
    
    // Application Details
    public DateTime ApplicationDate { get; set; } = DateTime.UtcNow;
    public string Status { get; set; } = "Pending"; // Pending, Approved, Rejected, Under Review
    public string? Remarks { get; set; }
    public int? ReviewedBy { get; set; } // UserId
    public DateTime? ReviewedDate { get; set; }

    // Documents (URLs or file paths)
    public string? PhotoUrl { get; set; }
    public string? BirthCertificateUrl { get; set; }
    public string? PreviousMarksheetUrl { get; set; }
    public string? AddressProofUrl { get; set; }
}

File: SchoolSync_SpringBoot/backend/src/main/java/com/schoolsync/entity/Admission.java
package com.schoolsync.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "admissions")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Admission {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long admissionId;

    @Column(nullable = false, unique = true)
    private String applicationNumber;

    // Personal Details
    @Column(nullable = false)
    private String firstName;

    @Column(nullable = false)
    private String lastName;

    @Column(nullable = false)
    private LocalDate dateOfBirth;

    @Column(nullable = false)
    private String gender;

    @Column(nullable = false)
    private String email;

    @Column(nullable = false)
    private String phone;

    @Column(nullable = false)
    private String address;

    @Column(nullable = false)
    private String city;

    @Column(nullable = false)
    private String state;

    @Column(nullable = false)
    private String pincode;

    // Academic Information
    @Column(nullable = false)
    private String appliedClass;

    @Column(nullable = false)
    private String academicYear;

    // Parent/Guardian Information
    @Column(nullable = false)
    private String fatherName;

    @Column(nullable = false)
    private String fatherOccupation;

    @Column(nullable = false)
    private String fatherPhone;

    @Column(nullable = false)
    private String motherName;

    @Column(nullable = false)
    private String motherOccupation;

    @Column(nullable = false)
    private String motherPhone;

    private String guardianName;
    private String guardianRelation;
    private String guardianPhone;

    // Previous School Information
    private String previousSchool;
    private String previousClass;
    private Double previousMarks;

    // Application Status
    @Column(nullable = false)
    private String status; // Pending, Under Review, Approved, Rejected

    @Column(nullable = false)
    private LocalDateTime applicationDate;

    private String remarks;

    // Roll Number (generated when approved)
    @Column(unique = true)
    private String rollNumber;
}
